### DISCUSSION TITLE
Use custom component as children of PopoverContent

### DISCUSSION THREAD

**Discussion by gamegee** [chakra-ui/chakra-ui#2774] [Q&A]
I'm implementing custom component on top of chakra-ui `Popover`

But for some reason, I am not able to trigger the open event from custom `PopoverContent` component.
I thought using `forwardRef` should be enough, but it is not working 🤔 

Here is an example,
[Sandbox link](https://codesandbox.io/s/weathered-haze-cd6gk?file=/src/app.jsx:276-280)

```tsx
import React from "react";
import {
  Button,
  forwardRef,
  Popover,
  PopoverContent,
  PopoverTrigger
} from "@chakra-ui/core";

const MyButton = forwardRef((props, ref) => {
  return (
    <Button ref={ref} h="s">
      Custom MyButton with forwardRef
    </Button>
  );
});

export default function App() {
  return (
    <>
     {/* Won't trigger the popover open*/} 
      <h1>Custom</h1>
      <Popover>
        <PopoverTrigger>
          <MyButton />
        </PopoverTrigger>
        <PopoverContent>Content</PopoverContent>
      </Popover>


      {/* Trigger the popover open*/} 
      <h1>Default button</h1>
      <Popover>
        <PopoverTrigger>
          <Button h="s">Button</Button>
        </PopoverTrigger>
        <PopoverContent>Content</PopoverContent>
      </Popover>
    </>
  );
}
```

Not sure what I am doing wrong ?

**Comment by thinusn** [comment]
Also ran into this, see https://github.com/chakra-ui/chakra-ui/issues/3121#issuecomment-766065952 wrapping it in a `Box` => `<Box><MyButton /></Box>` makes it work.

**Comment by Swiftwork** [comment]
Even though this is old, I wanted to mention that your example works fine @gamegee, you just missed spreading props to your custom button (which contains the onClick toggle method).
```jsx
const MyButton = forwardRef((props, ref) => {
  return (
    <Button ref={ref} h="s" {...props}>
      Custom MyButton with forwardRef
    </Button>
  );
});
```
